core/lib/dal/src/transactions_web3_dal.rs

```diff
@@ -151,6 +151,29 @@ impl TransactionsWeb3Dal<'_, '_> {
             .await
     }
 
+    pub async fn get_unstable_transaction_execution_info(
+        &mut self,
+        hash: H256,
+    ) -> DalResult<Option<StorageTransactionExecutionInfo>> {
```

Comment:
Bikeshedding: I'd try not to return storage models from public API (a bit questionable here given that this API is marked as unstable). Instead, it could make sense to return an `api` type (most of the methods in `_Web3Dal` do that), or `serde_json::Value`.